src/sntp.rs

```diff
@@ -174,6 +195,17 @@ impl EspSntp {
             (*tv).tv_usec,
         );
     }
+
+    unsafe extern "C" fn sync_cb_for_subscribe(tv: *mut timeval) {
+        let duration: core::time::Duration = core::time::Duration::from_secs((*tv).tv_sec as u64);
```

Comment:
Move the duration calculations inside the `if` statement below. t is pointless to compute `duration` if nobody is subscribed.